### PR TITLE
Fix report schedules for reports with missing timezone

### PR DESCRIPTION
**Pull request by akselsjogren**
With this patch we will not assume that there is a 'report_timezone'
report option stored, which is the case for reports that were saved
before the 'report_timezone' option was introduced.

Log to Ninja log on notice level (not enabled by default) when sending
reports, so that it can be tracked easier whether we're trying to send
reports when scheduled or not.

Signed-off-by: Aksel Sjögren <asjogren@itrsgroup.com>

This fixes MON-11912.

### CODE REVIEW

File: modules/reports/models/scheduled_reports.php [hunk 1]
```diff
@@ -337,25 +337,25 @@ static function get_period_schedules($period_str)
...
+		$default_timezone = date_default_timezone_get();
 
 		$db = Database::instance();
 
 		$sql = <<<'SQL'
 		SELECT sr.*, rp.periodname, opt.value AS timezone
 		FROM scheduled_reports sr
 		INNER JOIN scheduled_report_periods rp ON rp.id = sr.period_id
-		INNER JOIN saved_reports_options opt ON opt.report_id = sr.report_id
+		LEFT JOIN saved_reports_options opt ON opt.report_id = sr.report_id
```
Comment (jacobbaungard) [inline modules/reports/models/scheduled_reports.php]: My SQL skills are kinda rusty, but I do not really understand why this change is here.

Before I assume we got the timezone option from `saved_reports_options` if it existed, and I guess that has not changed with a `LEFT JOIN` or?

Comment (akselsjogren) [reply modules/reports/models/scheduled_reports.php]: With an `INNER JOIN`, there has to be a match for the `ON` clause for the row to be returned (i.e. `ON opt.report_id = sr.report_id AND opt.name = 'report_timezone'`. With `LEFT JOIN`, if the ON clause doesn't match for the right table ("saved_report_options"), it will still return the row, but filled with NULL for the missing values from right table.
So effectively, if there was no `report_timezone` option for some reason, with an INNER JOIN, the schedule wouldn't get returned at all.

Comment (jacobbaungard) [reply modules/reports/models/scheduled_reports.php]: Ahh, ok. Makes sense
